src/app/data-planes/components/DataPlaneDetails.vue

```diff
@@ -158,38 +158,80 @@
     <template #xds-configuration>
       <KCard>
         <template #body>
-          <EnvoyData
-            data-path="xds"
-            :mesh="dataplaneOverview.mesh"
-            :dpp-name="dataplaneOverview.name"
-            query-key="envoy-data-data-plane"
-          />
+          <DataSource
+            v-slot="{ data, error, refresh }: EnvoyDataSource"
+            :src="`/meshes/${props.dataplaneOverview.mesh}/dataplanes/${props.dataplaneOverview.name}/data-path/xds`"
+          >
+            <ErrorBlock
+              v-if="error"
+              :error="error"
+            />
+
+            <LoadingBlock v-else-if="data === undefined" />
+
+            <EmptyBlock v-else-if="data === ''" />
+
+            <EnvoyData
+              v-else
+              :content="data"
+              query-key="envoy-data-xds-data-plane"
+              @refresh="refresh"
+            />
+          </DataSource>
```

Comment:
Why do we need to extract the data fetching from the `EnvoyData` component. It just seems very verbose and redundant. Seems like we could achieve the same with just passing the :src as a prop to `EnvoyData` no?

Reply:
That’s possible. I’ve made a change for this. The downside to that is how we source the `EnvoyDataSource` type. I currently picked one at random because they’re virtually the same between dataplanes and zones, but it’s weird to import either inside this generic component.